src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs

```diff
@@ -18,14 +19,28 @@ private SourceInfo(string markdown, string file, int lineNumber)
 
         public int LineNumber { get; }
 
+        public int ValidLineCount { get; }
+
         public static SourceInfo Create(string markdown, string file, int lineNumber = 1)
         {
-            return new SourceInfo(markdown, file, lineNumber);
+            return new SourceInfo(markdown, file, lineNumber, GetValidLineCount(markdown));
         }
 
         public SourceInfo Copy(string markdown, int lineOffset = 0)
         {
-            return new SourceInfo(markdown, File, LineNumber + lineOffset);
+            return new SourceInfo(markdown, File, LineNumber + lineOffset, GetValidLineCount(markdown));
+        }
+
+        private static int GetValidLineCount(string markdown)
+        {
+            var validLineCount = 0;
+            var trimEndNewLine = markdown.Trim('\n');
+            for (var i = trimEndNewLine.Length - 1; i >= 0; i--)
+            {
+                if (trimEndNewLine[i] == '\n')
```

Comment:
`\r\n` has been replaced with `\r` in `Normalize` method

---
In reply to: [76769024](https://github.com/dotnet/docfx/pull/646#discussion_r76769024) [](ancestors = 76769024,76760278)